xpdan/data_reduction.py

```diff
@@ -140,6 +140,136 @@ def construct_event_stream(self, header):
 xpd_data_proc = DataReduction()
 ai = AzimuthalIntegrator()
 
+
+def image_stream(events, handler):
+    for e in events:
+        yield e[handler.image_field]
+
+
+def associate_dark(header, events, handler):
+    dark_uid = header.start.get(an_glbl.dark_field_key, None)
+    if dark_uid is None:
+        print("INFO: no dark frame is associated in this header, "
+              "subrraction will not be processed")
+        dark_img = None
+    else:
+        dark_search = {'uid': dark_uid}
+        dark_header = handler.exp_db(**dark_search)
+        dark_img = np.asarray(handler.exp_db.get_images(dark_header,
+                                                        handler.image_field)
+                              ).squeeze()
+    for e in events:
+        yield dark_img
+
+
+def subtract_gen(event_stream1, event_stream2):
```

Comment:
I think we need docstrings.   Keep it brief where it is blindingly obvious, as this case, but the docstrings make it easier to interrogate these functions at runtime (for example this yields event_stream1 - event_stream2 and not vice versa)